### PR TITLE
Fixes multivariate division

### LINKED ISSUE #3105: Multivariate division algorithm produces quotients q_i such that summands q_if_i has bigger degree than original polynomial f
**Describe the bug**
When dividing a polynomial $f$ by polynomials $f_1, \ldots, f_n$, the multivariate division algorithm returns quotients $q_i$ such that $q_if_i$ has bigger degree then $f$.

**To Reproduce**
The example to reproduce can also be found in "Ideals, Varieties and Algorithms", Ch. 2 Exercise §3.5
```
julia> R, (x,y,z) = QQ[:x, :y, :z]
(Multivariate polynomial ring in 3 variables over QQ, QQMPolyRingElem[x, y, z])

julia> f = x^3 - x^2*y - x^2*z + x
x^3 - x^2*y - x^2*z + x

julia> f1 = x^2*y - z
x^2*y - z

julia> f2 = x*y - 1
x*y - 1

julia> Q, r = reduce_with_quotients(f, [f1, f2], ordering = deglex(R))
([x^2 -x^3-x], 0)
```

But then the total degree of Q[1]*f1 (and Q[2]*f2) is bigger than the total degree of f. 

```
julia> total_degree(Q[1]*f1)
5

julia> total_degree(f)
3
```
An thus we obviously also have $lm_{prec}(f) \prec lm_{\prec}(Q[1]*f1)$ with respect to graded lexicographic order.

**Expected behavior**
We would expect that the quotient Q gives us $q_i f_i$ such that $tdeg(q_i f_i) < tdeg(f)$ for $i = 1,2$, see for example Theorem 3 in Ch.2, §3 of "Ideals, Varieties and Algorithms" by Cox et Al.. According to the documentation (https://docs.oscar-system.org/stable/CommutativeAlgebra/GroebnerBases/groebner_bases/#Division-With-Remainder) we would expect $lm_{prec}(q_i*f_i) \prec lm_{prec}(f_i)$.

**System:**
```
julia> Oscar.versioninfo(full=true)
OSCAR version 0.13.0
  combining:
    AbstractAlgebra.jl   v0.31.1
    GAP.jl               v0.9.8
    Hecke.jl             v0.20.3
    Nemo.jl              v0.35.3
    Polymake.jl          v0.11.8
    Singular.jl          v0.18.16
  building on:
    Antic_jll               v0.201.500+0
    Arb_jll                 v200.2300.0+0
    Calcium_jll             v0.401.100+0
    FLINT_jll               v200.900.7+0
    GAP_jll                 v400.1200.200+7
    Singular_jll            v403.208.800+0
    libpolymake_julia_jll   v0.11.0+0
    libsingular_julia_jll   v0.40.3+0
    polymake_jll            v400.1100.0+0
See `]st -m` for a full list of dependencies.

Julia Version 1.9.0
Commit 8e630552924 (2023-05-07 11:25 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin22.4.0)
  CPU: 4 × Intel(R) Core(TM) m3-6Y30 CPU @ 0.90GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, skylake)
  Threads: 1 on 4 virtual cores
Official https://julialang.org/ release
```

### PR DESCRIPTION
**Pull request by ederc**
This PR fixes #3105.

`reduce_with_...` now uses an equivalent algorithm as `reduce` and fixes wrong behaviour when reducing w.r.t. to a non-GB.

### CODE REVIEW

**Comment by hannes14** [comment]
Project.toml needs Singular = "0.22.3" instead of Singular = "0.22.0"

**Comment by thofma** [comment]
Since no one rebuilt the libsingular_julia binaries, Singular 0.22.3 does not contain https://github.com/oscar-system/Singular.jl/commit/a7a074994e02d5718fbdcf8d4495fbebe67877bd, that is, "divrem2".

**Comment by hannes14** [comment]
see https://github.com/JuliaPackaging/Yggdrasil/pull/8138

**Comment by lgoettgens** [comment]
I think you need a new Singular.jl release containing https://github.com/oscar-system/Singular.jl/pull/786, and then bump the compat in the Project.toml again (probably to 0.22.4 then)

File: Project.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@ Preferences = "1"
 Random = "1.6"
 RandomExtensions = "0.4.3"
 Serialization = "1.6"
-Singular = "0.22.0"
+Singular = "0.22.3"
```
Comment (lgoettgens) [inline Project.toml]:
```suggestion
Singular = "0.22.4"
```

**Review by fingolfin — APPROVED** [review]

**Review by wdecker — APPROVED** [review]

**Pull request merged**